specs/CookieManagement.md

```diff
@@ -0,0 +1,491 @@
+  [propget] HRESULT Expires([out, retval] double* expires);
+  /// Set the Expires property. Cookies are session cookies
+  /// and will not be persistent if Expires is negative.
+  [propput] HRESULT Expires([in] double expires);
+
+  /// Whether this cookie is http-only.
+  /// True if a page script or other active content cannot access this
+  /// cookie. The default is false.
+  [propget] HRESULT IsHttpOnly([out, retval] BOOL* httpOnly);
+  /// Set the HttpOnly property.
+  [propput] HRESULT IsHttpOnly([in] BOOL httpOnly);
+
+  /// SameSite status of the cookie which represents the enforcement mode of the cookie.
+  /// The default is COREWEBVIEW2_COOKIE_SAME_SITE_KIND_LAX.
+  [propget] HRESULT SameSite([out, retval] COREWEBVIEW2_COOKIE_SAME_SITE_KIND* sameSite);
+  /// Set the SameSite property.
+  [propput] HRESULT SameSite([in] COREWEBVIEW2_COOKIE_SAME_SITE_KIND sameSite);
+
+  /// The security level of this cookie. True if the client is only to return
+  /// the cookie in subsequent requests if those requests use HTTPS.
+  /// The default is false.
+  /// Note that cookie that requests COREWEBVIEW2_COOKIE_SAME_SITE_KIND_NONE but
+  /// is not marked Secure will be rejected.
+  [propget] HRESULT IsSecure([out, retval] BOOL* secure);
+  /// Set the Secure property.
+  [propput] HRESULT IsSecure([in] BOOL secure);
+}
+
+/// Creates, adds or updates, gets, or or view the cookies. The changes would
+/// apply to the context of the user profile. That is, other WebViews under the
+/// same user profile could be affected.
+[uuid(588C8A15-A28A-4FFD-926B-5E6EE7449E7C), object, pointer_default(unique)]
+interface ICoreWebView2CookieManager : IUnknown {
+  /// Create a cookie object with a specified name, value, domain, and path.
+  /// One can set other optional properties after cookie creation.
+  /// This only creates a cookie object and it is not added to the cookie
+  /// manager until you call AddOrUpdateCookie.
+  /// See ICoreWebView2Cookie for more details.
+  HRESULT CreateCookieWithDetails(
+    [in] LPCWSTR name,
+    [in] LPCWSTR value,
+    [in] LPCWSTR domain,
+    [in] LPCWSTR path,
+    [out, retval] ICoreWebView2Cookie** cookie);
+
+  
+  /// Creates a cookie whose params matches those of the specified cookie.
+  HRESULT CreateCookie(
+    [in] ICoreWebView2StagingCookie* cookieParam,
+    [out, retval] ICoreWebView2StagingCookie** cookie);
+
+  /// Gets a list of cookies matching the specific URI.
+  /// You can modify the cookie objects, call
+  /// ICoreWebView2CookieManager::AddOrUpdateCookie, and the changes
+  /// will be applied to the webview.
+  HRESULT GetCookies(
+    [in] LPCWSTR uri,
+    [in] ICoreWebView2GetCookiesCompletedHandler* handler);
+
+  /// Adds or updates a cookie with the given cookie data; may overwrite
+  /// equivalent cookies if they exist.
+  HRESULT AddOrUpdateCookie([in] ICoreWebView2Cookie* cookie);
+
+  /// Deletes a cookie whose params matches those of the specified cookie.

```

Comment:
Is it the same params as above used to match and delete: domain, path, & name? If the value is different it will still be deleted? If its a subset of the properties of the cookie that are used to delete we need to explicitly say which properties are used.